ambientweather2sqlite/database.py

```
@@ -178,140 +92,268 @@
+
+        cursor = self.conn.cursor()
+        cursor.execute(query)
 
         result: list[dict[str, float | int | str] | None] = [None for _ in range(24)]
         for row in cursor:
             result[int(row["hour"])] = dict(row)
 
         return result
+
+# Global database manager instance
+db_manager: DatabaseManager | None = None
```

Flake8 (7.2.0)

[error] expected 2 blank lines after class or function definition, found 1

(E305)